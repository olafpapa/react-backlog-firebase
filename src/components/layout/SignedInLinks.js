import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { signOut } from '../../store/actions/authActions'

export const SignedInLinks = (props) => {
  const handleSignOut = (e) => {
    e.preventDefault()
    props.signOut()
  }

  return (
    <ul className="right">
      <li><NavLink to='/create'>New Project</NavLink></li>
      {/*eslint-disable-next-line*/}
      <li><a onClick={ handleSignOut }>Log out</a></li>
      <li><NavLink to='/' className="btn btn-floating pink lighten-1">
        { props.profile.initials }
      </NavLink></li>
    </ul>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    signOut: () => dispatch(signOut())
  }
}
export default connect(null, mapDispatchToProps)(SignedInLinks)
