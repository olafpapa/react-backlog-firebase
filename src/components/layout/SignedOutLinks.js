import * as React from 'react';
import { NavLink } from 'react-router-dom'

export const SignedOutLinks = () => {
  return (
    <ul className="right">
      <li><NavLink to='/signup'>Sign Up</NavLink></li>
      <li><NavLink to='/signin'>Login</NavLink></li>
    </ul>
  );
};

export default SignedOutLinks
