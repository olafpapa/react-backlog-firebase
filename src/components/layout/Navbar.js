import React from 'react'
import SignedInLinks from './SignedInLinks'
import SignedOutLinks from './SignedOutLinks'
import { connect } from 'react-redux'

export const Navbar = (props) => {
  const { auth, profile } = props
  const links = auth.uid ? <SignedInLinks profile={profile}/> : <SignedOutLinks/>
  return (
    <nav className="nav-wrapper blue darken-3">
      <div className="container">
        { links }
      </div>
    </nav>

  );
};

const mapStateToProps = (state) => {
  return {
    auth: state.firebase.auth,
    profile: state.firebase.profile,
  }
}

export default connect(mapStateToProps)(Navbar)
