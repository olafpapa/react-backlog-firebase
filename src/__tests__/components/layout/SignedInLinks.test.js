import React from 'react'
import { BrowserRouter, NavLink } from 'react-router-dom'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { SignedInLinks } from '../../../components/layout/SignedInLinks'

describe('<SignedInLinks />', () => {

  const profile = {
    initials: 'AA'
  }
  let wrapper;

  // LOG OUTのモック
  const mockSignOut = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SignedInLinks profile={ profile } signOut={ mockSignOut }/>)
  })

  it('LOG OUTボタン押下', () => {
    wrapper.find('a').simulate('click', {
      preventDefault() {}
    })
    // ボタンが1回押されたか
    expect(mockSignOut).toHaveBeenCalledTimes(1)
  });

  it('子コンポーネントが存在すること', () => {
    // NavLinkコンポーネントの数を取得し、2であればOK
    expect(wrapper.find(NavLink).length).toBe(2);
  });

  test('<SignedInLinks />のスナップショット', () => {
    const tree = renderer
      .create(<BrowserRouter><SignedInLinks profile={ profile } signOut={ mockSignOut }/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
