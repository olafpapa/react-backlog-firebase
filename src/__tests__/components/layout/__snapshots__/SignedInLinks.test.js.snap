// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignedInLinks /> <SignedInLinks />のスナップショット 1`] = `
<ul
  className="right"
>
  <li>
    <a
      aria-current={null}
      href="/create"
      onClick={[Function]}
    >
      New Project
    </a>
  </li>
  <li>
    <a
      onClick={[Function]}
    >
      Log out
    </a>
  </li>
  <li>
    <a
      aria-current="page"
      className="btn btn-floating pink lighten-1 active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      AA
    </a>
  </li>
</ul>
`;
